tensordict/nn/common.py

```diff
@@ -166,8 +269,8 @@ def wrapper(
                                 f"but is expected to execute that function."
                             )
                 tensordict = make_tensordict(tensordict_values)
-                out = func(_self, tensordict, *args, **kwargs)
-                out = tuple(out[key] for key in _self.out_keys)
+                out = func(_self, tensordict, **kwargs)
```

Comment:
Why no `*args` here? Is it not possible that `source` has fewer entries than `args` and that we don't exhaust it?

Reply:
I made the change, but it's a bit confusing IMO.
Imagine this scenario:
```python
        class MyModuleNest(nn.Module):
            in_keys = [("a", "c"), "d"]
            out_keys = ["b"]

            @dispatch(separator="_")
            def forward(self, tensordict, other):
                tensordict["b"] = tensordict["a", "c"] + tensordict["d"] + other
                return tensordict

        module = MyModuleNest()
        other = 1
        (b,) = module(torch.zeros(1, 2), torch.ones(1, 2), other)
        assert (b == 2).all()
```
Now say you want to use `d` as a kwarg:
```python
        class MyModuleNest(nn.Module):
            in_keys = [("a", "c"), "d"]
            out_keys = ["b"]

            @dispatch(separator="_")
            def forward(self, tensordict, other):
                tensordict["b"] = tensordict["a", "c"] + tensordict["d"] + other
                return tensordict

        module = MyModuleNest()
        other = 1
        (b,) = module(torch.zeros(1, 2), other, d=torch.ones(1, 2), )
        assert (b == 2).all()
```
That breaks but with not apparent reason (it's not super clear from the function signature).

Reply:
That's not great, but I can create the same problem with a `nn.Module` no?

```python
class MyModule(nn.Module):
    def forward(self, a, d, other):
        pass

module = MyModule()
other = 1
module(torch.zeros(10), other, d=torch.ones(10))
```

Isn't this just a case of the user providing arguments to the function incorrectly? If I want to use `other` as a positional argument, then I necessarily need to supply both `a` and `d` as positional. 

In any case, in your example don't we get an error from [this line](https://github.com/pytorch-labs/tensordict/pull/256/files#diff-731e98da172c0a815c68a2e01d1e3ba85a59a9b0bec820aff620537fd76b303aR259)?